common-app-lib/src/cmr/common_app/services/search/elastic_search_index.clj

```diff
@@ -48,39 +49,96 @@
+            (re-find #"Malformed scrollId" body))
+      (errors/throw-service-error :bad-request (format "Invalid scroll id [%s]" scroll-id))
+      (throw ex))))
+
+(defmethod handle-es-exception 404
+  [ex scroll-id]
+  (if scroll-id
+    (errors/throw-service-error :not-found (format "Scroll session [%s] does not exist" scroll-id))
+    (throw ex)))
+
+(defmethod handle-es-exception :default
+  [ex _]
+  (throw ex))
+
+(defn- scroll-search 
+  "Performs a scroll search, handling errors where possible."
+  [context scroll-id]
+  (try
+    (esd/scroll (context->conn context) scroll-id :scroll (es-config/elastic-scroll-timeout))
+    (catch clojure.lang.ExceptionInfo e
+           (handle-es-exception e scroll-id))))
+
+(defn- do-send
+  "Sends a query to ES, either normal or using a scroll query."
+  [context index-info query]
+  (if-let [scroll-id (:scroll-id query)]
+    (scroll-search context scroll-id)
+    (esd/search (context->conn context) (:index-name index-info) [(:type-name index-info)] query)))
+      
+(defn- send-query
+  "Send the query to ES using either a normal query or a scroll query. Handle socket exceptions
+  by retrying."
+  [context index-info query]
+  (transmit-conn/handle-socket-exception-retries
+    (do-send context index-info query)))
+
 (defmulti send-query-to-elastic
   "Created to trace only the sending of the query off to elastic search."
   (fn [context query]
     (:page-size query)))
 
 (defmethod send-query-to-elastic :default
   [context query]
-  (let [{:keys [page-size offset concept-type aggregations highlights]} query
-        elastic-query (q2e/query->elastic query)
-        sort-params (q2e/query->sort-params query)
+  (let [elastic-query (q2e/query->elastic query)
+        {sort-params :sort-params 
+         aggregations :aggs 
+         highlights :highlight :as execution-params} (query->execution-params query)
+        concept-type (:concept-type query)
         index-info (concept-type->index-info context concept-type query)
-        fields (query-fields->elastic-fields
-                concept-type
-                (or (:result-fields query) (concept-type+result-format->fields concept-type query)))
-        from offset
-        query-map (util/remove-nil-keys (merge elastic-query
-                                               {:version true
-                                                :sort sort-params
-                                                :size page-size
-                                                :from from
-                                                :fields fields
-                                                :aggs aggregations
-                                                :highlight highlights}))]
+        query-map (-> elastic-query
+                      (merge execution-params)
+                      util/remove-nil-keys)]
     (info "Executing against indexes [" (:index-name index-info) "] the elastic query:"
```

Comment:
Probably a good idea. I didn't want to add it here since this is at the info level and the scroll-id is huge, but scrolling probably not going to  be that common, so it probably won't clog up our logs.